pkg/backporter/httpcache_test.go

```diff
@@ -0,0 +1,179 @@
+			expectedValue: []byte("Test string"),
+			isCached:      true,
+		},
+		{
+			name:     "Key is absent",
+			cache:    map[string][]byte{"key": []byte("Test string")},
+			key:      "absent",
+			isCached: false,
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			bc := newBugzillaCache()
+			bc.cache = tc.cache
+			cachedVal, isCached := bc.get(tc.key)
+			if tc.isCached != isCached {
+				t.Errorf("entry mismatch -  expected: %v, got: %v", tc.isCached, isCached)
+			}
+			if isCached {
+				if !bytes.Equal(tc.expectedValue, cachedVal) {
+					t.Errorf("cached value wrong expected: %s, got: %s", tc.expectedValue, cachedVal)
+				}
+			}
+		})
+	}
+}
+
+type fakeTransport struct {
+	response *http.Response
+	err      error
+}
+
+func (t fakeTransport) RoundTrip(req *http.Request) (resp *http.Response, err error) {
+	return t.response, t.err
+}
+
+func TestRoundTrip(t *testing.T) {
+	bodyText := "body text"
+	resp := &http.Response{
+		Status:     http.StatusText(http.StatusOK),
+		StatusCode: http.StatusOK,
+		Body:       ioutil.NopCloser(bytes.NewBuffer([]byte(bodyText))),
+	}
+	body, err := httputil.DumpResponse(resp, true)
+	if err != nil {
+		t.Fatalf("failed to serialize dummy response: %v", err)
+	}
+	testcases := []struct {
+		name     string
+		fake     fakeTransport
+		cache    map[string][]byte
+		isCached bool
+	}{
+		{
+			name: "Get payload directly from server",
+			fake: fakeTransport{
+				response: resp,
+			},
+			cache:    map[string][]byte{},
+			isCached: false,
+		},
+		{
+			name: "Get cached response",
+			fake: fakeTransport{
+				response: resp,
+			},
+			cache:    map[string][]byte{"http://somewhere.com/": body},
+			isCached: true,
+		},
+		{
+			name: "Error request to fake server",
+			fake: fakeTransport{
+				err: errors.New("fake error"),
+			},
+			cache:    map[string][]byte{},
+			isCached: false,
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			r, err := http.NewRequest("GET", "http://somewhere.com/", nil)
+			if err != nil {
+				t.Fatalf("failed to make request to fake server: %v", err)
+			}
+			bc := newBugzillaCache()
+			bc.cache = tc.cache
+			tp := cachingTransport{
+				cache:     bc,
+				transport: tc.fake,
+			}
+			resp, err := tp.RoundTrip(r)
+			if err != tc.fake.err {
+				t.Errorf("wrong error - expected %v, got %v", tc.fake.err, err)
+			}
+			if err != nil {
+				return
+			}
+			if resp.StatusCode != tc.fake.response.StatusCode {
+				t.Errorf("incorrect status code - expected %v, got %v", tc.fake.response.StatusCode, resp.StatusCode)
+			}
+			defer resp.Body.Close()
+			body, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				t.Fatalf("failed to parse response body: %v", err)
+			}
+			if string(body) != bodyText {
+				t.Errorf("incorrect body - expected %v, got %v", bodyText, string(body))
+			}
+			if tc.isCached {
+				respFromCache := resp.Header.Get(xFromCache)
+				if respFromCache != "1" {
```

Comment:
@GeoBK I can not unresolve the conversation above but did you see the second coment? That one is still open

Reply:
Yeah, I saw that 👍

Reply:
I think this can be better expressed as:
```
if isRespFromCache := resp.Header.Get(xFromCache) == "1"; isResponseFromCache != tc.isCached {
  t.Errorf("Expected resp to be from cache: %t, response was from cache: %t", tc.isCached, isResponseFromCache)
}
```